go/vt/vttablet/tabletmanager/rpc_replication.go

```diff
@@ -520,6 +522,23 @@ func (tm *TabletManager) demotePrimary(ctx context.Context, revertPartialFailure
 	}
 	defer tm.unlock()
 
+	finishCtx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	go func() {
+		select {
+		case <-finishCtx.Done():
+		// Finished running DemotePrimary. Nothing to do.
+		case <-time.After(10 * topo.RemoteOperationTimeout):
+			// We waited for over 10 times of remote operation timeout, but DemotePrimary is still not done.
+			// Collect more information and signal demote primary is indefinitely stalled.
+			log.Errorf("DemotePrimary seems to be stalled. Collecting more information.")
+			tm.QueryServiceControl.SetDemotePrimaryStalled()
```

Comment:
Yes, exactly!

Reply:
if we read the end of `demotePrimary` and we have called `SetDemotePrimaryStalled`, what is the correct course of action? it seems like we're assuming this will never happen. should we do something like block forever without returning, to ensure that the tablet doesn't accidentally make forward progress or re-enter the set of serving tablets?

Reply:
> even then I don't really see an issue with the tablet rejoining the serving tablets, until it is eventually restarted by the operator.

hm I think that is potentially a problem. because if the operator gets notified that a tablet is stalled, it's going to forcefully throw that tablet away with the assumption that (a) there is another tablet that is tablet that is the real primary and (b) the stalled primary is not serving any traffic. if the stalled primary is able to rejoin the set of serving tablets, both of those assumptions go out the window, and it is unsafe for the operator to safely throw away the tablet.

Reply:
Okay! I added few more safeties to ensure nothing goes wrong - 
1. After we set `DemotePrimaryStalled` we immediately trigger a health check update, which would make vtgate mark this tablet not-serving and not send it any requests ever again, because we never clear the field.
2. For any requests already sent, if `DemotePrimaryStalled` is set, we won't process it on vttablet and instead just return an error.

I think with these safeguards we can be sure htat a vttablet is not going to accept any new writes once we mark it as stalled. 

WDYT @maxenglander? Let's also wait for @deepthi to be able to take a look.